### DISCUSSION TITLE
/shell parameters not working (server side support issue)

### DISCUSSION THREAD

**Discussion by zhuyaliang** [FreeRDP/FreeRDP#10241] [General]
```xfreerdp /shell:"C:\Windows\notepad.exe"``` does not work
I tried using the master branch, but the problem still persists. I suspect the problem is with Windows, but I don't know how to configure Windows to make this command work
This problem exists in Windows 10, 11, and Server 2016. Who can help me? I am not familiar with Windows at all

**Comment by akallabeth** [comment]
@zhuyaliang this option is equivalent to the [alternate shell](https://learn.microsoft.com/en-us/windows-server/remote/remote-desktop-services/clients/rdp-files) option of a `rdp` file.

how to enable that on server side I also don´t know though.

**Reply by zhuyaliang** [reply]
I edited the RDP file
```
alternate shell:s:"C:\Windows\notepad.exe"
```
This still doesn't work

**Comment by ygm521** [comment]
linux used xfreerdp run cmd on remote windows： **does not work！**
./xfreerdp /u:xxxx /p:xxxx /v:172.18.10.115:3389 /drive:local,/tmp/ /size:1024x768 /title:test /cert:ignore /shell:"cmd.exe /c ipconfig > Z:\\ipconfig.txt"

**Comment by ygm521** [comment]
old xfreerdp: https://warroom.rsmus.com/scripting-rdp/

**Comment by akallabeth** [comment]
as it seems to be hard to find: https://stackoverflow.com/questions/1226772/can-rdp-clients-launch-remote-applications-and-not-desktops
